dataservice/__init__.py

```diff
@@ -119,10 +118,10 @@ def register_error_handlers(app):
     NB: Exceptions to be handled must be imported in the head of this module
     """
     from dataservice.api import errors
-    app.register_error_handler(HTTPException, errors.http_error)
     app.register_error_handler(IntegrityError, errors.integrity_error)
-    app.register_error_handler(404, errors.http_error)
-    app.register_error_handler(400, errors.http_error)
+    from werkzeug.exceptions import default_exceptions
+    for ex in default_exceptions:
+        app.register_error_handler(ex, errors.http_error)
```

Comment:
No luck with HTTPException? It looks like default_exceptions is just all the subclasses of HTTPException, so you would think it would work.

Reply:
I think we have to go with this approach for now. Doesn't look like the fix for the issue with HTTPException is in Flask 0.12. Its slated for Flask milestone 1.0 which isn't complete yet

https://github.com/pallets/flask/pull/2314